src/librustc_privacy/lib.rs

```diff
@@ -1504,9 +1517,29 @@ impl<'a, 'tcx: 'a> SearchInterfaceForPrivateItemsVisitor<'a, 'tcx> {
                 self.tcx.lint_node(lint::builtin::PRIVATE_IN_PUBLIC, node_id, self.span,
                                    &format!("{} (error {})", msg, err_code));
             }
+
         }
+
         false
     }
+
+    /// An item is 'leaked' from a private dependency if all
+    /// of the following are true:
+    /// 1. It's contained within a public type
+    /// 2. It does not come from a crate marked as public
+    fn leaks_private_dep(&self, item_id: DefId) -> bool {
+        // Never do any leak checking if the feature is not enabled
+        if !self.tcx.features().public_private_dependencies {
+            return false
```

Comment:
Feature gates generally do not *change* behavior, but rather gate it.
(Otherwise we'd get one behavior before stabilization and another after, that would be something else than "stabilization".)

I would expect this logic to start working if `--extern-public` was passed (which in its turn *is* gated).
`--extern-public=""` would mean "no public deps, but run the leak checking".
Then path to stabilization would be clear - allow `--extern-public` on stable.

Reply:
Would it make sense to remove the feature entirely? From what I've seen, features aren't used to gate flags. Should I gate it behind `-Z unstable-options`?

Reply:
>`-Z unstable-options`

Yes, AFAIK that's how command line flags are gated, and the "in-source" feature gate can be removed if it's not used.

Reply:
I agree with @petrochenkov that we probably don't want this clause here. I think we'll instead want an allow-by-default lint with an unstable CLI flag.

That way testing would both enable the lint and then also pass the unstable CLI flags as necessary